### PR TITLE
Fixed missing class method when multiple trait aliases are used

### PR DESCRIPTION
**Pull request by kukulich**
Fixes #1453

Yes, the code is ugly but the PHP feature is ugly too... I really hope I will not see any other trait bug in near future!

### CODE REVIEW

File: src/Reflection/ReflectionClass.php [hunk 1]
```diff
@@ -1366,18 +1377,36 @@ static function (ReflectionClass $trait): string {
...
+
+        $traits       = $this->getTraits();
+        $traitAliases = [];
+
+        foreach ($traits as $trait) {
+            if (! array_key_exists($trait->getName(), $this->traitsData['aliases'])) {
+                continue;
+            }
+
+            foreach ($this->traitsData['aliases'][$trait->getName()] as $traitAliasDefinition) {
```
Comment (Ocramius) [inline src/Reflection/ReflectionClass.php]: I'm quite worried about nesting multiple `foreach` blocks here: the code complexity is exploding quite a bit

File: src/Reflection/ReflectionClass.php [hunk 2]
```diff
@@ -1438,37 +1467,37 @@ private function computeTraitsData(ClassNode|InterfaceNode|TraitNode|EnumNode $n
...
-                $traitsData['hashes'][$lowerCasedMethodHash] = $methodHash;
-
-                if ($adaptation instanceof Node\Stmt\TraitUseAdaptation\Alias) {
-                    if ($adaptation->newModifier !== null) {
-                        /** @var int-mask-of<ReflectionMethodAdapter::IS_*> $modifier */
-                        $modifier                                       = $adaptation->newModifier;
-                        $traitsData['modifiers'][$lowerCasedMethodHash] = $modifier;
+                $usedTraits = $adaptation->trait !== null ? [$adaptation->trait] : $traitNames;
+
+                foreach ($usedTraits as $usedTrait) {
```
Comment (Ocramius) [inline src/Reflection/ReflectionClass.php]: Same remarks about foreach complexity

File: src/Reflection/ReflectionClass.php [hunk 3]
```diff
@@ -1418,7 +1447,7 @@ public function getTraitAliases(): array
      *   //   ['MyTrait1::foo' => 'MyTrait2::foo']
      *
      * @return array{
-     *     aliases: array<non-empty-string, non-empty-string>,
+     *     aliases: array<non-empty-string, list<array{alias: non-empty-string, method: non-empty-string, hash: non-empty-string}>>,
```
Comment (Ocramius) [inline src/Reflection/ReflectionClass.php]: From a consumer of this method perspective, what is this hash?

File: test/unit/Fixture/TraitFixture.php [hunk 4]
```diff
@@ -135,3 +135,20 @@ class TraitFixtureF
     }
 
 }
+
+trait Trait1FixtureG {
```
Comment (Ocramius) [inline test/unit/Fixture/TraitFixture.php]: IMO worth moving to a string source locator within the test: it's only used within that one test, and shouldn't be mixed with the rest

**Review by Ocramius — APPROVED** [review]
The test additions and clarity thereof is fantastic : my only concerns are with code complexity, and this code is exploding with them.

I would solit this part, even at major performance costs, just ti keep it manageable.

Traits are an edge case in good design: they are a problem that keeps haunting us, and I'd like to push them in their own "naughty" corner.

File: src/Reflection/ReflectionClass.php (hunk 3 above)
Comment (kukulich) [reply src/Reflection/ReflectionClass.php]: It's more described above - the hash was already there before but now I need more data about aliases.

File: test/unit/Fixture/TraitFixture.php (hunk 4 above)
Comment (kukulich) [reply test/unit/Fixture/TraitFixture.php]: Done.

File: src/Reflection/ReflectionClass.php (hunk 1 above)
Comment (kukulich) [reply src/Reflection/ReflectionClass.php]: I've just tried to refactored it a little.

File: src/Reflection/ReflectionClass.php (hunk 2 above)
Comment (kukulich) [reply src/Reflection/ReflectionClass.php]: I've just tried to refactored it a little.

**Review by Ocramius — APPROVED** [review]
Thanks @kukulich, this looks much much better!

**Comment by Ocramius** [comment]
🚢 thanks @kukulich!

**Pull request merged**